src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,60 @@
-export const Pagination = () => {};
+import React from 'react';
+import classNames from 'classnames';
+import {
+  getLinkDataCy,
+  getLinkHref,
+  getLinkTextContent,
+  handleClickPagination,
+} from './pagination-utils';
+
+export interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesAmount = Math.ceil(total / perPage);
+
+  return (
+    <ul
+      className="pagination"
+      onClick={(event: React.MouseEvent<HTMLUListElement>) =>
```

Comment:
Inline functions within the render method can be inefficient as they are created on each render. Consider moving this function outside the component and passing necessary arguments, or memoizing it with `useCallback` if it depends on props or state.